File: backend/app/openliga/models/table.py
"""
Pydantic models for OpenLigaDB table responses.

Endpoint: /getbltable/{league}/{season}
"""

from pydantic import BaseModel, Field


class OpenLigaTeamStanding(BaseModel):
    """Single team from /getbltable/{league}/{season}"""

    team_id: int = Field(alias="teamInfoId")
    team_name: str = Field(alias="teamName")
    short_name: str = Field(alias="shortName")
    team_icon_url: str = Field(alias="teamIconUrl")

    points: int
    matches: int
    won: int
    draw: int
    lost: int
    goals: int
    opponent_goals: int = Field(alias="opponentGoals")
    goal_diff: int = Field(alias="goalDiff")

    model_config = {"populate_by_name": True}

File: frontend/src/types/table.ts
/**
 * Types for Bundesliga table endpoint.
 */

export interface TeamStanding {
  position: number
  team_name: string
  short_name: string
  team_icon_url: string
  points: number
  matches: number
  won: number
  draw: number
  lost: number
  goals: number
  opponent_goals: number
  goal_diff: number
}

export interface TableResponse {
  season: string
  standings: TeamStanding[]
}
